### PR TITLE
Tool to translate geometries

### PR DESCRIPTION
**Pull request by rinkk**

### CODE REVIEW

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+// ThirdParty
+#include "tclap/CmdLine.h"
+
+// ThirdParty/logog
+#include "logog/include/logog.hpp"
+
+// BaseLib
+#include "LogogSimpleFormatter.h"
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: Full include path...

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+// MathLib
+#include "MathLib/Vector3.h"
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: Do we need the comment about a MathLib include for a MathLib include?

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#include "FileIO/XmlIO/Qt/XmlGmlInterface.h"
+
+// GeoLib
+#include "GeoLib/GEOObjects.h"
+
+#include <QtGui/QApplication>
+
+int main(int argc, char *argv[])
+{
+	QApplication app(argc, argv);
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: I don't know too much about qt. Can you explain, why this app, which is not used later, is needed?

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	logogCout->SetFormatter(*custom_format);
+
+	TCLAP::CmdLine cmd("Moves the points of a geometry by a given displacement vector.", ' ', "0.1");
+	TCLAP::ValueArg<double> z_arg("z", "z", "displacement in z direction", false, 0.0, "z-displacement");
+	cmd.add(z_arg);
+	TCLAP::ValueArg<double> y_arg("y", "y", "displacement in y direction", false, 0.0, "y-displacement");
+	cmd.add(y_arg);
+	TCLAP::ValueArg<double> x_arg("x", "x", "displacement in x direction", false, 0.0, "x-displacement");
+	cmd.add(x_arg);
+	TCLAP::ValueArg<std::string> geo_output_arg("o","output", "output geometry file (*.gml)", true, "", "output file");
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: Long lines are difficult to read. http://ufz.github.io/styleguide/cppguide.xml#Line_Length

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	cmd.add(geo_output_arg);
+	TCLAP::ValueArg<std::string> geo_input_arg("i","input", "input geometry file (*.gml)", true, "", "input file");
+	cmd.add(geo_input_arg);
+	cmd.parse( argc, argv );
+
+	GeoLib::GEOObjects geo_objects;
+	FileIO::XmlGmlInterface xml(geo_objects);
+	if (xml.readFile(geo_input_arg.getValue()))
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: If readFile fails, there is no error message and the return value of main is 0, which prohibits this tool being used in scripts.

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+			displacement[1] = y_arg.getValue();
+		if (z_arg.isSet())
+			displacement[2] = z_arg.getValue();
+
+		std::vector<std::string> geo_names;
+		geo_objects.getGeometryNames(geo_names);
+
+		std::vector<GeoLib::Point*> const* point_vec = geo_objects.getPointVec(geo_names[0]);
+		std::size_t const n_points = point_vec->size();
+		for (std::size_t i=0; i<n_points; ++i)
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: `for (auto point : *point_vec)`
`i` and `n_points` are then unnecessary.

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+			for (std::size_t c=0; c<3; ++c)
+				(*(*point_vec)[i])[c] += displacement[c];
+	
+		std::string out_file_name(geo_output_arg.getValue());
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: Unused variable

File: Applications/Utils/GeoTools/MoveGeometry.cpp [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+			displacement[0] = x_arg.getValue();
+		if (y_arg.isSet())
+			displacement[1] = y_arg.getValue();
+		if (z_arg.isSet())
+			displacement[2] = z_arg.getValue();
+
+		std::vector<std::string> geo_names;
+		geo_objects.getGeometryNames(geo_names);
+
+		std::vector<GeoLib::Point*> const* point_vec = geo_objects.getPointVec(geo_names[0]);
```
Comment (endJunction) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: Unsure here, but could geo_names be an empty vector?

File: Applications/Utils/GeoTools/MoveGeometry.cpp (hunk 3 above)
Comment (rinkk) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: The XML classes require an existing QApplication to work.

File: Applications/Utils/GeoTools/MoveGeometry.cpp (hunk 8 above)
Comment (rinkk) [inline Applications/Utils/GeoTools/MoveGeometry.cpp]: it cannot be a nullptr. if the vector were empty nothing bad would happen...

**Comment by endJunction** [comment]
:fast_forward:

**Pull request merged**